browser_patches/webkit/patches/bootstrap.diff

```diff
@@ -7220,6 +7236,54 @@ index 3ecf866005da6da9cec7c7930e6a4f93cb39d6e0..d81dc9ac0c433e00ab4ad73e20685609
      {
          RELEASE_ASSERT_NOT_REACHED();
      }
+diff --git a/Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp b/Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp
+index 8789f11c3643999c30dfdb1c9b939ae45bb0a51f..422961d2b43e6da9e36df4cdeb06fe91ac57a644 100644
+--- a/Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp
++++ b/Source/WebCore/platform/win/ClipboardUtilitiesWin.cpp
+@@ -38,6 +38,7 @@
+ #include <wtf/text/CString.h>
+ #include <wtf/text/StringBuilder.h>
+ #include <wtf/text/win/WCharStringExtras.h>
++#include "Pasteboard.h"
+ 
+ #if USE(CF)
+ #include <CoreFoundation/CoreFoundation.h>
+@@ -724,7 +725,10 @@ template<typename T> void getStringData(IDataObject* data, FORMATETC* format, Ve
+     STGMEDIUM store;
+     if (FAILED(data->GetData(format, &store)))
+         return;
+-    dataStrings.append(String(static_cast<T*>(GlobalLock(store.hGlobal)), ::GlobalSize(store.hGlobal) / sizeof(T)));
++    // The string here should be null terminated, but it could come from another app so lets lock it
++    // to the size to prevent an overflow.
++    String rawString = String(static_cast<T*>(GlobalLock(store.hGlobal)), ::GlobalSize(store.hGlobal) / sizeof(T));
++    dataStrings.append(String::fromUTF8(rawString.utf8().data()));
```

Comment:
Is there a reason to create intermediate String rather than just call String::fromUTF8 on the buffer? (there is an overload that takes (LChar*, size_t))

Reply:
yeah. I pass through utf8 to set the right size if its null terminated.